packages/react/src/components/Dialog/Dialog.test.tsx

```diff
@@ -233,6 +234,83 @@ test('should set alignment on dialog footer', () => {
   expect(footer).toHaveClass('text--align-right');
 });
 
+describe('when "forceAction" is false', () => {
+  test('it should call "onClose" when escape is pressed', async () => {
+    const user = userEvent.setup();
+    const onClose = jest.fn();
+    render(
+      <Dialog {...defaultProps} onClose={onClose} forceAction={false}>
+        Test Dialog
+      </Dialog>
+    );
+
+    expect(onClose).not.toHaveBeenCalled();
+    await user.keyboard('{Escape}');
+    expect(onClose).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('when "forceAction" is true', () => {
+  test('it should not call "onClose" when escape is pressed', async () => {
+    const user = userEvent.setup();
+    const onClose = jest.fn();
+    render(
+      <Dialog {...defaultProps} onClose={onClose} forceAction={true}>
+        Test Dialog
+      </Dialog>
+    );
+
+    expect(onClose).not.toHaveBeenCalled();
+    await user.keyboard('{Escape}');
+    expect(onClose).not.toHaveBeenCalled();
+  });
+});
+
+describe('when a tooltip is rendered within the dialog', () => {
+  const renderTooltipInDialog = () => {
+    const CustomTooltipDialog = () => {
+      const [show, setShow] = useState(true);
+
+      return (
+        <Dialog
+          heading={'Tooltip Dialog'}
+          show={show}
+          onClose={() => setShow(false)}
+        >
+          <TooltipTabstop tooltip="Hello Tooltip">show tooltip</TooltipTabstop>
+        </Dialog>
+      );
+    };
+
+    const user = userEvent.setup();
+
+    const doc = render(<CustomTooltipDialog />);
```

Comment:
Nit: Instead of using the return value from `render`, we typically use `screen`, example:

```tsx
render(<Component />)
screen.getByRole(...)
```